core/src/test/java/com/linecorp/armeria/server/logging/AccessLogFormatsTest.java

```diff
@@ -255,6 +276,114 @@ public void requestLogAvailabilityException() {
         log.endResponse();
     }
 
+    @Test
+    public void requestLogComponent() {
+        final DummyRequestContext ctx = new DummyRequestContext();
+        final DefaultRequestLog log = spy(new DefaultRequestLog(ctx));
+
+        List<AccessLogComponent> format;
+
+        log.startRequest(channel, SessionProtocol.H2, "www.example.com");
+        log.requestHeaders(HttpHeaders.of(HttpMethod.GET, "/armeria/log"));
+
+        final Instant requestStartTime = Instant.ofEpochMilli(1000000000L);
+        final Duration duration = Duration.ofMillis(1000000000L);
+
+        log.endRequest();
+        when(log.requestStartTimeMillis()).thenReturn(requestStartTime.toEpochMilli());
+        when(log.requestDurationNanos()).thenReturn(duration.toNanos());
+
+        format = AccessLogFormats.parseCustom("%{requestStartTimeMillis}L " +
+                                              "%{requestEndTimeMillis}L " +
+                                              "%{requestDurationMillis}L");
+        assertThat(AccessLogger.format(format, log)).isEqualTo(
+                String.join(" ",
+                            String.valueOf(requestStartTime.toEpochMilli()),
+                            String.valueOf(requestStartTime.plus(duration).toEpochMilli()),
+                            String.valueOf(duration.toMillis())));
+
+        format = AccessLogFormats.parseCustom("\"%{requestCause}L\"");
+        assertThat(AccessLogger.format(format, log)).isEqualTo("\"-\"");
+        when(log.requestCause()).thenReturn(new IllegalArgumentException("detail_message"));
+        assertThat(AccessLogger.format(format, log)).isEqualTo("\"detail_message\"");
```

Comment:
Shouldn't we also include the exception type? e.g. `IllegalArgumentException: detail_message`

Reply:
Good point, it would be better!